03week/ticTacToe.js

```diff
@@ -3,91 +3,155 @@
+    return ((board[0][0] === playerOne && board[0][1] === playerOne && board[0][2] === playerOne) || (board[1][0]
+        === playerOne && board[1][1] === playerOne && board[1][2] === playerOne) || (board[2][0] === playerOne
+        && board[2][1] === playerOne && board[2][2] === playerOne) || (board[0][0] === playerTwo && board[0][1]
+        === playerTwo && board[0][2] === playerTwo) || (board[1][0] === playerTwo && board[1][1] === playerTwo &&
+        board[1][2] === playerTwo) || (board[2][0] === playerTwo && board[2][1] === playerTwo && board[2][2] === playerTwo))
 }
 
 function verticalWin() {
-  // Your code here
+    return ((board[0][0] === playerOne && board[1][0] === playerOne && board[2][0] === playerOne) || (board[0][1]
+        === playerOne && board[1][1] === playerOne && board[2][1] === playerOne) || (board[0][2] === playerOne
+        && board[1][2] === playerOne && board[2][2] === playerOne) || (board[0][0] === playerTwo && board[1][0]
+        === playerTwo && board[2][0] === playerTwo) || (board[0][1] === playerTwo && board[1][1] === playerTwo &&
+        board[2][1] === playerTwo) || (board[0][2] === playerTwo && board[1][2] === playerTwo && board[2][2] === playerTwo))
 }
 
 function diagonalWin() {
-  // Your code here
+    return ((board[0][0] === playerOne && board[1][1] === playerOne && board[2][2] === playerOne) || (board[2][0]
+        === playerOne && board[1][1] === playerOne && board[0][2] === playerOne) || (board[0][0] === playerTwo && board[1][1]
+        === playerTwo && board[2][2] === playerTwo) || (board[2][0] === playerTwo && board[1][1] === playerTwo &&
+        board[0][2] === playerTwo))
+
 }
 
 function checkForWin() {
-  // Your code here
+    return (diagonalWin() || verticalWin() || horizontalWin())
+
+}
+// Check for a valid spot on the board
+//   Only accept intigers, 0-2
+//Check for empty space
+//   Not already filled with 'X' or 'O'
+//If both are true then place a marker.
+// Player one is “X” will start by entering a position based on the board and then go to player'O'
+// This position will no longer be a valid move.
+// If same position is tried a message will be given of “..........”
+// Check for win
+//  Check for win horizontal
+//  check for win vertical
+// Check diagonal win
+// If no win then change player turn.
+// if win. return You Won.. ==> reset board.
+//
+
+const userInputIsValid = (row, column) => {
+    return (row < 3 && row >= 0) && (column < 3 && column >= 0)
+}
+
+const isSpaceEmpty = (row, column) => {
+    const space = board[row][column];
+    return (space !== 'X' && space !== 'O')
+}
+const placeAMarker = (row, column) => {
+    if (isSpaceEmpty(row, column) === true){
+        return board[row][column] = playerTurn
+        // console.log(board)
+    }else {
+        console.log("This is not a valid play.");
+    }
 }
 
 function ticTacToe(row, column) {
-  // Your code here
+    if (userInputIsValid(row, column) && isSpaceEmpty(row, column)) {
```

Comment:
This is so solid, it's very readable, and it's right in your main function.
:+1: